lib/elastic_apm/spies/mongo.rb

```diff
@@ -91,14 +91,13 @@ def push_event(event)
               context: build_context(event)
             )
 
-          events[event.operation_id] = span
+          events << span
         end
 
         def pop_event(event)
-          span = events.delete(event.operation_id)
           return unless (curr = ElasticAPM.current_span)
 
-          curr == span && ElasticAPM.end_span
+          curr == events[-1] && ElasticAPM.end_span(events.pop)
```

Comment:
Do we need a Mutex around this line?

Reply:
That's a good question. The `events` list is a thread-local variable so we wouldn't need a mutex to protect mutating that, however, what could happen, is one thread sets `curr` in line 98 and then another thread compares `curr` to its thread local `event` stack.
So we'll need a mutex around the whole thing, to make sure the current span is being compared to the correct thread's `events` stack.

Reply:
Update: the current span is a thread local variable so we don't need a mutex at all.

Reply:
Yes, I was thinking that another span might get pushed during `&&` in this line, right?

Reply:
No, because the `events` list is local to a thread. So another thread can't possibly push to another thread's `events` list.

Reply:
Right 👍